spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainerIntegration2Tests.java

```diff
@@ -258,6 +271,54 @@ public void testMissingListener() throws Exception {
 		assertTrue(containerStoppedForAbortWithBadListener());
 	}
 
+	@Test
+	public void testRestartConsumerOnBasicQosIoException() throws Exception {
+		this.template.convertAndSend(queue.getName(), "foo");
+
+		ConnectionFactory connectionFactory = new SingleConnectionFactory("localhost", BrokerTestUtils.getPort());
+
+		final AtomicBoolean networkGlitch = new AtomicBoolean();
+
+		class MockChannel extends PublisherCallbackChannelImpl {
+
+			public MockChannel(Channel delegate) {
+				super(delegate);
+			}
+
+			@Override
+			public void basicQos(int prefetchCount) throws IOException {
+				if (networkGlitch.compareAndSet(false, true)) {
+					throw new IOException("Intentional connection reset");
+				}
+				super.basicQos(prefetchCount);
+			}
+
+		}
+
+		Connection connection = spy(connectionFactory.createConnection());
+		when(connection.createChannel(anyBoolean())).then(new Answer<Channel>() {
+
+			@Override
+			public Channel answer(InvocationOnMock invocation) throws Throwable {
+				return new MockChannel((Channel) invocation.callRealMethod());
+			}
+
+		});
+
+		DirectFieldAccessor dfa = new DirectFieldAccessor(connectionFactory);
+		dfa.setPropertyValue("connection", connection);
+
+		CountDownLatch latch = new CountDownLatch(1);
+		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(connectionFactory);
+		container.setMessageListener(new MessageListenerAdapter(new PojoListener(latch)));
+		container.setQueueNames(queue.getName());
+		container.afterPropertiesSet();
```

Comment:
Nice idea to subclass the `PCCI` for the mock.

LGTM; adding `setRecoveryInterval(500)` to speed up the test and merging.